doc/languages.rst

```diff
@@ -949,6 +949,18 @@ to view the docstring of the syntax checker.  Likewise, you may use
          A list of additional library directories. Relative paths are relative
          to the buffer being checked.
 
+   .. syntax-checker:: rust-cargo-clippy
+
+      Check syntax and lint with `cargo clippy <https://github.com/Manishearth/rust-clippy>`_.
+
+      .. note::
+
+         This syntax checker requires ``cargo clippy``, and therefore a nighlty version of rustc.
```

Comment:
Please also extend the paragraph above after `.. supported-language:: rust` to mention the new syntax checker and how it relates with the existing rust checkers.